src/virtual-machine.js

```diff
@@ -1033,6 +1034,25 @@ class VirtualMachine extends EventEmitter {
         return null;
     }
 
+    /**
+     * Reorder target by index. Return whether a change was made.
+     * @param {!string} targetIndex Index of the target.
+     * @param {!number} newIndex index that the target should be moved to.
+     * @returns {boolean} Whether a target was reordered.
+     */
+    reorderTarget (targetIndex, newIndex) {
+        targetIndex = MathUtil.clamp(targetIndex, 0, this.runtime.targets.length - 1);
+        newIndex = MathUtil.clamp(newIndex, 0, this.runtime.targets.length - 1);
+        if (targetIndex === newIndex) return false;
+        const target = this.runtime.targets[targetIndex];
+        let targets = this.runtime.targets;
```

Comment:
You reference `this.runtime.targets` a few times before this line. You can move this declaration up.